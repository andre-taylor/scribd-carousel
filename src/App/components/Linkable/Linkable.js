import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import styles from './Linkable.module.css';

export const Linkable = ({ href, className, children }) => {
  return (
    <a href={href} className={cx(styles.linkableContainer, className)}>
      {children}
    </a>
  );
};

Linkable.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  href: PropTypes.string.isRequired,
};

export default Linkable;
