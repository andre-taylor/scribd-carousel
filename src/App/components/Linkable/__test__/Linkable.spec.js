import React from 'react';
import renderer from 'react-test-renderer';

import Linkable from '../Linkable';

describe('<Linkable />', () => {
  it('renders correctly with optional and required props', () => {
    const component = renderer.create(
      <Linkable href={'www.scribd.com'} className={'linkable'}>
        <div>Linkable container</div>
      </Linkable>
    );
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with all required props', () => {
    const component = renderer.create(
      <Linkable href={'www.scribd.com'}>
        <div>Linkable container</div>
      </Linkable>
    );
    expect(component).toMatchSnapshot();
  });
});
