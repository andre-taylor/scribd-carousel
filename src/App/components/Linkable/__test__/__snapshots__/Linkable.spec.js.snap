// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Linkable /> renders correctly with all required props 1`] = `
<a
  className="linkableContainer"
  href="www.scribd.com"
>
  <div>
    Linkable container
  </div>
</a>
`;

exports[`<Linkable /> renders correctly with optional and required props 1`] = `
<a
  className="linkableContainer linkable"
  href="www.scribd.com"
>
  <div>
    Linkable container
  </div>
</a>
`;
